### PR TITLE
Add 'ParseRockTheVoteReport' job.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request adds a `ParseRockTheVoteReport` job to parse the files that we've downloaded from `FetchRockTheVoteReport`. This should be the last step in moving this full flow into Northstar!

### How should this be reviewed?

This was extracted from [ImportFileRecords](https://github.com/DoSomething/chompy/blob/a52a23993bfb46a0f69ab27584cf784150ae0bb1/app/Jobs/ImportFileRecords.php) job in Chompy. By making these into separate jobs, we avoid complexity from one job (e.g. email subscriptions' `importOptions`) creeping into all the other possible imports. Shared logic for CSV parsing was extracted to a helper.

I'll make corresponding jobs for importing subscriptions & muting promotions as well.

### Any background context you want to provide?

I removed the `LogProgress` calls since we have run into trouble using Pusher on our free tier (due to usage caps). Since the majority of our uploads are kicked off by automated backend processes, I don't think this is worth continuing to do.

### Relevant tickets

References [Pivotal #178301643](https://www.pivotaltracker.com/story/show/178301643).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [ ] If new attributes were added to users, there is a corresponding PR to surface these in Aurora.
- [ ] If new attributes were added to users, then the data team already knows about these changes.

### CODE REVIEW

File: tests/Jobs/Imports/example-rtv-report.csv [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
+Complete,"user:60be61d276739310fb4b7b03,source:web,source_details:NewAccountCreationFlow",,,Success: 2902806,English,4/20/1994,runolfsdottir@example.is,Yes,Mr.,Mae,,Runolfsdottir,,32631 Bernhard Drive,,Jayneshire,Philadelphia,PA,55125,No,,,,,,,None (No Affiliation),Black or African American,555-528-3167,,No,No,No,No,,,,,No,No,,No,2021-05-21 19:44:06 -0400,No,No,No,Yes,,[],New,,2021-06-01 10:32:34 -0400,Yes,TRUE
+Step 3,"user:60be61d23de3eb3df64b7b04,source:web,source_details:profile",,,,English,1/21/2003,dmante@example.com,Yes,Mr.,Domenico,,Mante,,593 Stanton Corner,Apt. 488,Tristontown,,CA,123456,No,,,,,,,,,555-860-4848,Mobile,Yes,No,Yes,No,,,,,No,No,,No,2021-05-22 21:01:32 -0400,No,No,Yes,No,,,,,,No,FALSE
+Step 1,"user:60be61d271153745cc4b7b05,source:web,source_details:NewAccountCreationFlow",,,,English,,brooks06@example.org,No,,,,,,24685 Vandervort Bypass,Apt. 273,Yundtville,,CT,55123,No,,,,,,,,,555-336-0545,,Yes,No,Yes,No,,,,,No,No,,No,2021-05-21 11:20:04 -0400,No,No,Yes,No,,,,,,No,FALSE
```
Comment (DFurnes) [inline tests/Jobs/Imports/example-rtv-report.csv:5]: I pulled a few lines from a real report, and replaced PII (user IDs, names & emails, birthdates, addresses, etc.) with data generated by Faker. I tried to get a wide range of possible report statuses, but I'm sure there will be more outliers that we may want to add in here later on!

File: tests/Jobs/Imports/ParseRockTheVoteReportTest.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        // Create a faked record that we can safely read & delete:
+        $csv = file_get_contents('tests/Jobs/Imports/example-rtv-report.csv');
+        Storage::put('report.csv', $csv);
+
+        $this->forceDispatch(new ParseRockTheVoteReport('report.csv'));
+
+        // There are 4 rows in the example CSV, so we should dispatch 4 jobs!
+        Bus::assertDispatchedTimes(ImportRockTheVoteRecord::class, 4);
+
+        // Then, we should clean up fater ourselves.
```
Comment (katiecrane) [inline tests/Jobs/Imports/ParseRockTheVoteReportTest.php:29]: Typo!
```suggestion
        // Then, we should clean up after ourselves.
```

**Review by katiecrane — APPROVED** [review]
Looks good to me! flagged one typo in a comment

File: app/Jobs/Imports/ParseRockTheVoteReport.php [hunk 3]
```diff
@@ -3,25 +3,62 @@
...
+            'import_type' => ImportType::$rockTheVote,
+            'row_count' => iterator_count($records),
+            'user_id' => optional($this->user)->id,
+        ]);
+
+        foreach ($records as $record) {
+            ImportRockTheVoteRecord::dispatch($record, $importFile);
+        }
+
+        // Now that we've chomped, delete the import file.
```
Comment (weerd) [inline app/Jobs/Imports/ParseRockTheVoteReport.php:60]: 😋

File: tests/Jobs/Imports/ParseRockTheVoteReportTest.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+     * Test that this job can parse a Rock The Vote CSV report.
+     *
+     * @return void
+     */
+    public function testParsesReport()
+    {
+        Bus::fake();
+        Storage::fake();
+
+        // Create a faked record that we can safely read & delete:
```
Comment (weerd) [inline tests/Jobs/Imports/ParseRockTheVoteReportTest.php:20]: Not sure if typo but sounds funny!
```suggestion
        // Create a fake record that we can safely read & delete:
```

**Review by weerd — APPROVED** [review]
🌮 😋

**Pull request merged**